src/passes/Print.cpp

```diff
@@ -174,8 +169,8 @@ struct SExprType {
   SExprType(Type type) : type(type){};
 };
 
-static std::ostream& operator<<(std::ostream& o, const SExprType& localType) {
-  Type type = localType.type;
+static std::ostream& operator<<(std::ostream& o, const SExprType& sType) {
```

Comment:
I'm not very sure what is this method exactly for anymore.. It seems it was added in #2675 to support multivalue printing for local types without commas, like `(i32 i64)`, but it has since grown and its name also changed from `LocalType` to `SExprType`... Can you help me on when we just use `printType` and when we use this method?

Reply:
Sorry, yes, it is a little confusing. I've added comments in this PR which might help. But probably we should rename things too.

* `printTypeName` prints a "serialized" name, as a list of characters. It has no spaces, in particular. We use this to emit a name for any type, no matter what it contains. Say an array contains `(ref eq)` then we'd print some serialization of that into `[ref_eq]` or such.
* SExprType prints out a type in normal s-format. For `(ref eq)` it would be exactly that (with spaces etc.). But that can recursively call `printTypeName` for the sub-types that it is declared from, when it needs their "serialized" name.

Reply:
It would be good to standardize on either using `printXXX` functions or using wrapper structs with bespoke `operator<< implementations`, but I don't think that needs to happen here. (I prefer the latter fwiw.)